src/main/java/mate/academy/security/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package mate.academy.security.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.security.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+import org.apache.commons.validator.routines.EmailValidator;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        String emailNoWhitespace = email.strip();
+        String passwordNoWhitespace = password.strip();
```

Comment:
We do not need to use strip() here.

Reply:
Why? What do you suppose to do if user make a mechanical erratum with whitespaces during registration or login? If nothing is done, the application will handle incorrect input data with whitespaces.

Reply:
Yes, but this way you will save data that will be different from the data entered by the user. If you're getting so close to realism, you should check such cases and throw an error

Reply:
Fine, now it is clear.